### PR TITLE
Fix updating inappropriate fields on upsert.

### PR DESCRIPTION
**Pull request by ashercodeorg**

### CODE REVIEW

File: pegasus/test/test_form_helpers.rb [hunk 1]
```diff
@@ -57,7 +58,7 @@ class FormHelpersTest < SequelTestCase
     it 'inserts a new form' do
       row = insert_or_upsert_form(
         'HocSignup2017',
-        DEFAULT_DATA.merge(email: "#{SecureRandom.hex(8)}@example.com")
+        DEFAULT_DATA.merge(email_s: "#{SecureRandom.hex(8)}@example.com")
```
Comment (ashercodeorg) [inline pegasus/test/test_form_helpers.rb:61]: An unrelated bug fix surfaced by my refactoring the `insert_or_upsert_form` method.

File: pegasus/helpers/form_helpers.rb [hunk 2]
```diff
@@ -56,25 +56,25 @@ def insert_or_upsert_form(kind, data, options={})
...
-  if (kind =~ /HocSignup/) == 0
+  if kind.start_with? 'HocSignup'
+    row_for_update = row.dup
+    [:created_at, :created_ip, :secret].each {|key| row_for_update.delete key}
     num_rows_updated = DB[:forms].
-      where(email: row[:email], kind: kind, name: row[:name]).
-      update(row)
+      where(kind: kind, email: row_for_update[:email], name: row_for_update[:name]).
+      update(row_for_update)
+    return row_for_update if num_rows_updated > 0
```
Comment (aoby) [inline pegasus/helpers/form_helpers.rb:65]: I know this is out of scope for this PR, but I think this would be a lot cleaner with a refactor to generalize. 
The form helpers shouldn't know or care about any specific form types, let alone make decisions based on that. 

What if instead we (optionally) implement a set of methods, say `should_update?(data)` and `update(data)` which we will call if they exist for the form class? Note we already [have](https://github.com/code-dot-org/code-dot-org/blob/1bbf2fe5fd3e807206bf52c8afceb709d5a5c7df/pegasus/helpers/form_helpers.rb#L55) the form class and check if methods exist. Or even just the `should_update?`, with only HOC forms implementing it and always returning true.

Comment (ashercodeorg) [reply pegasus/helpers/form_helpers.rb:65]: Agreed. Assuming I don't get caught up with other stuff, I'll plan on doing so in another PR.

Comment (ashercodeorg) [reply pegasus/helpers/form_helpers.rb:65]: Aside, I don't like the code (data manipulation) duplication between `insert_and_upsert_form` and `update_form`, nor the structure of `insert_and_upsert_form`. That methods don't return what you want them to return will make the refactor slightly more complicated.
